core/testing/src/test/kotlin/app/k9mail/core/testing/TestClockTest.kt

```diff
@@ -0,0 +1,30 @@
+package app.k9mail.core.testing
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import kotlin.test.Test
+import kotlinx.datetime.Instant
+
+internal class TestClockTest {
```

Comment:
An implementation that only returns `Instant.DISTANT_FUTURE` satisfies both tests. 

Maybe have one test that asserts that `now()` returns the time the Clock was initialized with, and one that initializes the clock to one value, then changes it to another value and asserts that the changed value is returned.